cargo-miri/bin.rs

```diff
@@ -585,6 +614,34 @@ fn phase_cargo_rustc(args: env::Args) {
         // (Need to do this here as cargo moves that "binary" to a different place before running it.)
         info.store(&out_filename("", ".exe"));
 
+        // Rustdoc expects us to exit with an error code if the test is marked as `compile_fail`,
+        // so we need to run Miri with `MIRI_BE_RUSTC` for a check-only build.
+        if std::env::var_os("MIRI_CALLED_FROM_RUSTDOC").is_some() {
+            let mut cmd = miri();
+
+            // use our own sysroot
+            if !has_arg_flag("--sysroot") {
+                let sysroot = env::var_os("MIRI_SYSROOT")
+                    .expect("the wrapper should have set MIRI_SYSROOT");
+                cmd.arg("--sysroot").arg(sysroot);
+            }
+            
+            // don't go into "code generation" (i.e. validation)
```

Comment:
What do you mean by "calidation"? `MIRI_BE_RUSTC` would actually do proper code generation.

Reply:
I was under the impression that that's what the `--emit` argument effectively does here. I adjusted the comment to communicate this more clearly.

Reply:
"validation" is rather ambiguous about what is being validated; that's why my question was about. The new comment explains that, thanks!